core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/ThrottlingHandler.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.applicationinsights.channel.concrete.inprocess;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.apache.http.Header;
+
+import com.google.common.base.Strings;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicy;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicyManager;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+
+public class ThrottlingHandler implements TransmissionHandler {
+
+	private TransmissionPolicyManager transmissionPolicyManager;
+	private final static String RESPONSE_RETRY_AFTER_DATE_FORMAT = "E, dd MMM yyyy HH:mm:ss";
+	
+	public ThrottlingHandler(TransmissionPolicyManager policy)
+	{
+		this.transmissionPolicyManager = policy;
+	}
+	
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+	}
+
+	public boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
```

Comment:
Similar to previous does this method need to be public. More-ever can we create an abstraction for this too, like an interface because it seems like this is being used in different handler classes you created.

Reply:
Fixed this by using the package scope. Only classes defined in the packages can see the method.